src/commands/search_tx_optimizations.rs

```diff
@@ -0,0 +1,93 @@
+use anyhow::{bail, Result};
+use clap::Args;
+use sqlx::{PgPool, Row};
+
+#[derive(Debug, Args)]
+#[command(about = "Command to apply or drop search transaction optimizations in the archive database.")]
+pub struct SearchTxOptimizationsCommand {
+  /// URL to the archive database
+  #[arg(long, env = "MINAMESH_ARCHIVE_DATABASE_URL")]
+  archive_database_url: String,
+
+  /// Apply optimizations
+  #[arg(long, conflicts_with = "drop")]
+  apply: bool,
```

Comment:
Could use a nested subcommand instead of the boolean flags + `conflicts_with`. That being said, it's not super important; let's merge for now.